### DISCUSSION TITLE
How to check for a condition in a triggered function?

### DISCUSSION THREAD

**Discussion by wsw70** [custom-components/pyscript#360] [Q&A] [answered]
I have a function that is called from triggers in several places:

```python
def volets_time(when="", entity_id=None, howMuch=None, condition="on"):
    log.info(f"setting up volets for {entity_id} at {when} → {howMuch}")

    @time_trigger(f"once({when})")
    def volets():
        if condition == "on":
            cover.set_cover_position(entity_id=entity_id, position=howMuch)
            log.info(f"volets {entity_id} → {howMuch}")
        else:
            log.info(
                f"volets {entity_id} should go {howMuch}, but automation is switched off"
            )

    return volets
```

It does not really matter what it does, the key point is the condition `if condition == "on":`. My intent was to have binary switches that allow the automation or not.

The problem of course is that I would need to pass the binary switch to th function, and not only its current state, as I do it incorrectly now:

```python
triggers = [
    volets.volets_time(
        entity_id="cover.martin_volets",
        howMuch="100",
        when="11:00",
        condition=input_boolean.martin_volets_matin,
    ),
    volets.volets_time(
        entity_id="cover.martin_volets",
        howMuch="0",
        when="sunset+1h",
        condition=input_boolean.martin_volets_soir,
    ),
]
```

What would be the correct way to pass (for instance) `input_boolean.martin_volets_matin` so that its current state is checked when the function fires up (at a given time)?

**Comment by craigbarratt** [comment, marked as answer]
It seems you need to defer the fetching of the state variable value from trigger creation to trigger running.  The easiest way would be to make `condition` a string (eg, `"input_boolean.martin_volets_soir"`), and use `state.get()` inside the inner trigger function to get its current value (ie, replace`condition` with `state.get(condition)`).
